src/databricks/labs/ucx/source_code/linters/imports.py

```diff
@@ -204,10 +223,26 @@ def visit_call(self, node: Call):
             return
         is_append = func.attrname == "append"
         changed = node.args[0] if is_append else node.args[1]
-        if isinstance(changed, Const):
-            self.sys_path_changes.append(AbsolutePath(node, changed.value, is_append))
-        elif isinstance(changed, Call):
-            self._visit_relative_path(changed, is_append)
+        relative = False
+        if isinstance(changed, Call):
+            if not self._match_aliases(changed.func, ["os", "path", "abspath"]):
+                return
+            relative = True
+            changed = changed.args[0]
+        try:
+            all_inferred = changed.inferred()
+            for inferred in all_inferred:
+                self._visit_inferred(changed, inferred, relative, is_append)
+        except InferenceError:
+            self.sys_path_changes.append(UnresolvedPath(changed, changed.as_string(), is_append))
+
+    def _visit_inferred(self, changed: NodeNG, inferred: NodeNG, is_relative: bool, is_append: bool):
+        if inferred is Uninferable or not isinstance(inferred, Const):
+            self.sys_path_changes.append(UnresolvedPath(changed, changed.as_string(), is_append))
```

Comment:
```suggestion
            self.sys_path_changes.append(UnresolvedPath(changed, changed.as_string(), is_append))
    return
```
Can we explicitly return, so that branches won't fall through accidentally?

Reply:
done